src/node_http2_core.h

```diff
@@ -307,12 +306,6 @@ class Nghttp2Stream {
   inline ~Nghttp2Stream() {
 #if defined(DEBUG) && DEBUG
     CHECK_EQ(session_, nullptr);
-    CHECK_EQ(queue_head_, nullptr);
-    CHECK_EQ(queue_tail_, nullptr);
-    CHECK_EQ(data_chunks_head_, nullptr);
-    CHECK_EQ(data_chunks_tail_, nullptr);
-    CHECK_EQ(current_headers_head_, nullptr);
-    CHECK_EQ(current_headers_tail_, nullptr);
```

Comment:
These are supposed to check emptyness, right? That can still happen if we like.

Reply:
Yeah, but they're largely worthless checks. The `Nghttp2Stream` instances are pooled are only rarely actually destroyed so these checks aren't really helping us with much. I had them in there largely to test for correctness as I was writing the code.